site-service-client/src/main/java/uk/ac/ox/ndph/mts/siteserviceclient/SiteServiceClient.java

```diff
@@ -0,0 +1,103 @@
+package uk.ac.ox.ndph.mts.siteserviceclient;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.util.UriComponentsBuilder;
+import uk.ac.ox.ndph.mts.siteserviceclient.configuration.ClientRoutesConfig;
+import uk.ac.ox.ndph.mts.siteserviceclient.exception.RestException;
+import uk.ac.ox.ndph.mts.siteserviceclient.model.SiteDTO;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+@Service
+public class SiteServiceClient {
+
+    private final WebClient webClient;
+
+    private final RequestExecutor requestExecutor;
+
+
+    @Autowired
+    public SiteServiceClient(WebClient.Builder webClientBuilder,
+                             @Value("${site.service.uri}") String siteServiceUri,
+                             RequestExecutor requestExecutor) {
+        this.requestExecutor = requestExecutor;
+        this.webClient = webClientBuilder.baseUrl(siteServiceUri).build();
+    }
+
+    public static Consumer<HttpHeaders> noAuth() {
+        return (headers) -> {
+        };
+    }
+
+    public static Consumer<HttpHeaders> basicAuth(final String username, final String password) {
+        return (headers) -> headers.setBasicAuth(username, password);
+    }
+
+    public static Consumer<HttpHeaders> bearerAuth(final String token) {
+        return (headers) -> headers.setBearerAuth(token);
+    }
+
+
+    public boolean entityIdExists(final String siteId,
+                                  final Consumer<HttpHeaders> authHeaders)
+            throws RestException {
+        Objects.requireNonNull(siteId, ResponseMessages.ID_NOT_NULL);
+        try {
+            getById(siteId, authHeaders);
+        } catch (RestException ex) {
+            return false;
+        }
+        return true;
```

Comment:
I think this logic can produce unwanted results - any RestExpection translates to not found entity isn't always the case...

Reply:
Fixed, but I'm not sure about it... The fix is not very webclient "native", tell me WDYT